embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java

```diff
@@ -501,7 +505,8 @@ public TaskReport run(TaskSource taskSource,
             }
 
             List<ColumnGetter> getters = newColumnGetters(con, task, querySchema, pageBuilder);
-            try (BatchSelect cursor = con.newSelectCursor(builtQuery, getters, task.getFetchRows(), task.getSocketTimeout())) {
+            try (BatchSelect cursor = con.newSelectCursor(builtQuery, getters, task.getFetchRows(),
+                task.getSocketTimeout(), task.getPreviewSampleRows())) {
```

Comment:
By this, it seems like `preview_sample_rows` always affects, even if it's not in the preview mode. Checking `Exec.isPreview()` in each `*InputConnection` does not look reasonable to me.

I meant that : 
* `newSelectCursor` would receive `OptionalInt limit` in general for any case, either in the preview or non-preview mode.
* `AbstractJdbcInputPlugin`, the caller of `newSelectCursor`, would determine whether to limit or not. `Exec.isPreview()` would be checked in `AbstractJdbcInputPlugin`, not in `newSelectCursor`.

Reply:
Thanks @dmikurube Got your point. I updated it.